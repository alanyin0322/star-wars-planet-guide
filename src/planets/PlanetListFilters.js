import React from 'react';

export default class PlanetListFilters extends React.Component {

  render() {
    return (
      <div>
        <input type="text" value={this.props.filter} onChange={this.props.setFilter}/>
        <select value={this.props.sortBy} onChange={this.props.setSortBy} >
          <option value="name-a-to-z">Name: A to Z</option>
          <option value="name-z-to-a">Name: Z to A</option>
          <option value="population-high-to-low">Population: High to Low</option>
          <option value="population-low-to-high">Population: Low to High</option>
        </select>
      </div>
    );
  }
} 