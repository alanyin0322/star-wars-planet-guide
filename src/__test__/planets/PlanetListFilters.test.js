import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PlanetListFilters from '../../planets/PlanetListFilters';

describe('PlanetListFilters component', () => {
  test('PlanetListFilters matches snapshot', () => {
    const tree = shallow(<PlanetListFilters />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  describe('setFilter', () => {
    test('gets rendered correctly on filter input', () => {
      const onFilterChange = jest.fn();
      const wrapper = shallow(<PlanetListFilters setFilter={onFilterChange} />);
      expect(wrapper.find('#filter').prop('onChange')).toEqual(onFilterChange);
    });

    test('called with the correct arguments', () => {
      const event = {target: {value: 'al'}};
      const onFilterChangeSpy = jest.fn();
      const wrapper = shallow(<PlanetListFilters setFilter={onFilterChangeSpy}/>);
      wrapper.find('#filter').simulate('change', event);
      expect(onFilterChangeSpy).toHaveBeenCalledWith(event);
    });
  });

  describe('setSortBy', () => {
    test('gets rendered correctly on sort-by select', () => {
      const onSortByChange = jest.fn();
      const wrapper = shallow(<PlanetListFilters setSortBy={onSortByChange} />);
      expect(wrapper.find('#sort-by').prop('onChange')).toEqual(onSortByChange);
    });

    test('called with the correct arguments', () => {
      const event = {target: {value: 'population-high-to-low'}};
      const onSortByChangeSpy = jest.fn();
      const wrapper = shallow(<PlanetListFilters setSortBy={onSortByChangeSpy}/>);
      wrapper.find('#sort-by').simulate('change', event);
      expect(onSortByChangeSpy).toHaveBeenCalledWith(event);
    });
  });
});