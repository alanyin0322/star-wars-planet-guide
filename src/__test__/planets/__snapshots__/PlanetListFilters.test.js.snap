// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetListFilters component PlanetListFilters matches snapshot 1`] = `
<div
  className="form-inline"
>
  <div
    className="form-group col-6"
  >
    <label>
      Search planet
    </label>
    <input
      className="form-control"
      id="filter"
      type="text"
    />
  </div>
  <div
    className="form-group col-6"
  >
    <label>
      Sort by
    </label>
    <select
      className="form-control-lg"
      id="sort-by"
    >
      <option
        value="name-a-to-z"
      >
        Name: A to Z
      </option>
      <option
        value="name-z-to-a"
      >
        Name: Z to A
      </option>
      <option
        value="population-high-to-low"
      >
        Population: High to Low
      </option>
      <option
        value="population-low-to-high"
      >
        Population: Low to High
      </option>
    </select>
  </div>
</div>
`;
